zerver/forms.py

```diff
@@ -379,6 +383,15 @@ def clean(self) -> Dict[str, Any]:
                     params={'username': self.username_field.verbose_name},
                 )
 
+            if action_key:
+                try:
+                    email_change_object = get_object_from_key(action_key, Confirmation.EMAIL_CHANGE)
+                except ConfirmationKeyException:
+                    raise ValidationError("Invalid key")
```

Comment:
In a normal flow, this should not occur as we redirect the user to appropriate error page on clicking confirmation link with this code https://github.com/zulip/zulip/pull/15805/commits/8b5450c534d443474a4377f306d883e807bff0fa#diff-2a3f35796feeb6b0ded8f5b2eb2a58f0R78-R81

So, I thought just saying "Invalid key" would be enough.